### ISSUE TITLE
simple-git in windows machine

### ISSUE THREAD

**Issue by somudcis** [steveukx/git-js#49]
I'm trying to use npm simple-git in windows machine. But when I run it with node.js it shows me the following error message.

events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: spawn git ENOENT

Node version: v0.12.2
npm version: 2.7.4

**Comment by steveukx** [comment]
Hi, I don't have access to a windows machine to test this out - have you ensured that you are able to use `git` as a named binary from the command prompt / process that you are calling the node script?

**Comment by CianMclennan** [comment]
I'm experiencing this same issue on windows. git executable is in the path. 

Error: spawn git ENOENT
    at exports._errnoException (util.js:746:11)
    at Process.ChildProcess._handle.onexit (child_process.js:1053:32)
    at child_process.js:1144:20
    at process._tickCallback (node.js:355:11)
    at Function.Module.runMain (module.js:503:11)
    at startup (node.js:129:16)
    at node.js:814:3
Error: spawn git ENOENT
    at exports._errnoException (util.js:746:11)
    at Process.ChildProcess._handle.onexit (child_process.js:1053:32)
    at child_process.js:1144:20
    at process._tickCallback (node.js:355:11)
    at Function.Module.runMain (module.js:503:11)
    at startup (node.js:129:16)
    at node.js:814:3

**Comment by Tomkinen** [comment]
Same here. Windows server 2012 and NodeJS 4.x.x. Git is included in the path.

Error: spawn git ENOENT
    at exports._errnoException (util.js:870:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:441:9)
    at process._tickCallback (node.js:355:17)

**Comment by laggingreflex** [comment]
Works now? Worked for me. Win 10 node 6.5

**Comment by stephenlautier** [comment]
I've used this on window 10, node 7.6.0 and window7 (i believe I have similar node version) and it works on both.

**Comment by SebastianZimmer** [comment]
Does work for me on Win10, but not on Win7 with Node 8.1.3.

**Comment by ccoenen** [comment]
I've tried a few basic exampels on a windows 10 machine with node 10 and simple-git 1.9.5. I did not run into any problems, yet.

I'd say this should be closed.

**Comment by kihonkai** [comment]
I've had a different but I think related issue so I thought I'd explain here instead of opening a new issue.

I can use simple-git in Node 8.11.4 but I get an error when I use the push method.  The following is a simplified example of my code:

```
const simpleGit = require('simple-git')(config.gitclonePath);

simpleGit.add('./*')
  .commit('commit message here')
  .push('origin', 'master')

```
The add and commit work fine but I get the following error:

> git-receive-pack 'C:\**PATH to clone folder here**': git-receive-pack: command not found
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.

the push has failed.  If I run `git push` using the Git Bash, Windows cmd or PowerShell then it works fine.  

As far as I know, I installed Git for Windows normally.  I picked the "Use Git from the Windows Command Prompt" option during the installer which is default as far as I know.

Git is in the PATH environment variable with this folder: `C:\Program Files\Git\cmd`.  I found the `git-receive-pack.exe` in two folder:

>C:\Program Files\Git\mingw64\bin\git-receive-pack.exe
>C:\Program Files\Git\mingw64\libexec\git-core\git-receive-pack.exe

One other folder in PATH that is related to Git is `C:\Program Files\TortoiseGit\bin` but that doesn't have either the main `git.exe` or that `git-receive-pack.exe`.

One workaround is adding `C:\Program Files\Git\mingw64\bin` to the PATH, it works fine in Node.js then.  

I don't know what is different about running it on the command line (cmd, Git Bash or Powershell) compared to using simple-git but there appears to be something.  That something different could be on Windows in general or just on my computer.

If I get chance, I'll look into it a bit more.  I'm also going to be running this on a Linux system soon, I imagine that will work with no problems :)
